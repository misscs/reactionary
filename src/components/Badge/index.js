import React from 'react'
import { connect } from 'react-redux'

import Avatar from 'components/Avatar/Avatar'
import './Badge.css'

const Badge = ({name}) => {
  return (
    <article className="badge">
      <Avatar/>
      <div className="badge-content">{name}</div>
    </article>
  )
}

Badge.propTypes = {
  name: React.PropTypes.string
}

export default Badge
