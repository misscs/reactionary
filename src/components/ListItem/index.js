import React from 'react'

function ListItem(props) {
  return (
    <li>
      {props.item}
    </li>
  )
}

ListItem.propTypes = {
  item: React.PropTypes.any,
}

export default ListItem
