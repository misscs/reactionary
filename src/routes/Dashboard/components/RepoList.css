.repo-list-item {
  border-bottom: 1px solid #aaa;
  padding: 0.5rem;
}
