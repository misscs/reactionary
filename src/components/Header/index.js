import React, { PropTypes } from 'react'

import Badge from 'components/Badge'
import HeaderItem from './Item'
import './Header.css'

class Header extends React.PureComponent  {

  render () {

    const userActions = this.props.loggedIn ? (
        <HeaderItem>
          <Badge name={this.props.name} />
        </HeaderItem>
      ) : (
        <HeaderItem>
          Register
        </HeaderItem>

      )

    return (
      <header>
        <HeaderItem className="logo">
          <span>CC</span>
        </HeaderItem>

        { userActions }
      </header>
    )
  }
}


Header.propTypes = {
  loggedIn: React.PropTypes.bool.isRequired,
  isFetching: React.PropTypes.bool,
  userActions: React.PropTypes.node,
}


export default Header
