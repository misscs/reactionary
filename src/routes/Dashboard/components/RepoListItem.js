/**
 * RepoListItem
 *
 * Lists the name and the issue count of a repository
 */

import React from 'react'
import { connect } from 'react-redux'

import ListItem from 'components/ListItem'
import './RepoList.css'

export class RepoListItem extends React.PureComponent {

  render() {
    const item = this.props.item

    const content = (
      <article className="repo-list-item">
        <h3>
          {item.name}
        </h3>
        <p>{item.description}</p>
      </article>
    )

    // Render the content into a list item
    return (
      <ListItem key={`repo-list-item-${item.full_name}`} item={content} />
    )
  }
}

RepoListItem.propTypes = {
  item: React.PropTypes.object,
}

export default connect()(RepoListItem)
