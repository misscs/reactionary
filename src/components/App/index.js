import React from 'react'
import { connect } from 'react-redux'
import { createStructuredSelector } from 'reselect'

import Header from 'components/Header'

import {
  getLoggedIn,
  getIsFetching,
  getUsername
} from 'routes/Login/LoginSelectors'

import './App.css'

class App extends React.Component {
  render() {
    return (
      <div>
        <Header
          loggedIn={this.props.loggedIn}
          isFetching={this.props.isFetching}
          name={this.props.name}
        />
        <main id="container">
          {this.props.children}
        </main>
      </div>
    )
  }
}

App.propTypes = {
  children: React.PropTypes.node,
}

const mapStateToProps = createStructuredSelector({
  loggedIn: getLoggedIn(),
  isFetching: getIsFetching(),
  name: getUsername()
})

export default connect(mapStateToProps)(App)
