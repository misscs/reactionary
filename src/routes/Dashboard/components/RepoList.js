import React, { PropTypes } from 'react'

import List from 'components/List'
import { RepoListItem } from './RepoListItem'

function RepoList({ repos }) {

  if (repos !== false) {
    return <List items={repos} component={RepoListItem} />
  }

  return null
}

RepoList.propTypes = {
  repos: PropTypes.any
}

export default RepoList
