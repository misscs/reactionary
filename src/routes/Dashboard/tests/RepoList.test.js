import React from 'react'
import renderer from 'react-test-renderer'

import RepoList from 'routes/Dashboard/components/RepoList'
import { RepoListItem } from 'routes/Dashboard/components/RepoListItem'

describe('Dashboard comonents: <RepoList /> & <RepoListItem />', () => {
  let item
  let items

  beforeEach(() => {
    item = {
      name: 'posthtml-postcss-modules',
      description: 'PostCSS CSS Modules Plugin',
      full_name: 'emilos/posthtml-postcss-modulese'
    }

    items = [
      {
        name: 'posthtml-postcss-modules',
        description: 'PostCSS CSS Modules Plugin',
        full_name: 'emilos/posthtml-postcss-modulese'
      },
      {
        name: 'posthtml-postcss-modules',
        description: 'PostCSS CSS Modules Plugin',
        full_name: 'emilos/posthtml-postcss-modulese'
      }
    ]
  })

  describe('<RepoList />', () => {
    it('should render a list of repositories', () => {
      const component = renderer.create(
        <RepoList repos={items} />
      )

      const tree=component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('should not render when nothing is passed', () => {
      const component = renderer.create(
        <RepoList repos={false} />
      )

      const tree=component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('<RepoListItem />', () => {
    it('should render one repository list item', () => {
      const component = renderer.create(
        <RepoListItem item={item} />
      )

      const tree=component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
