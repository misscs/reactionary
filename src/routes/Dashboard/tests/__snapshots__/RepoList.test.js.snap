// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard comonents: <RepoList /> & <RepoListItem /> <RepoList /> should not render when nothing is passed 1`] = `null`;

exports[`Dashboard comonents: <RepoList /> & <RepoListItem /> <RepoList /> should render a list of repositories 1`] = `
<div
  className="gHWurn"
>
  <ul
    className="iKHmLl"
  >
    <li>
      <article
        className="repo-list-item"
      >
        <h3>
          posthtml-postcss-modules
        </h3>
        <p>
          PostCSS CSS Modules Plugin
        </p>
      </article>
    </li>
    <li>
      <article
        className="repo-list-item"
      >
        <h3>
          posthtml-postcss-modules
        </h3>
        <p>
          PostCSS CSS Modules Plugin
        </p>
      </article>
    </li>
  </ul>
</div>
`;

exports[`Dashboard comonents: <RepoList /> & <RepoListItem /> <RepoListItem /> should render one repository list item 1`] = `
<li>
  <article
    className="repo-list-item"
  >
    <h3>
      posthtml-postcss-modules
    </h3>
    <p>
      PostCSS CSS Modules Plugin
    </p>
  </article>
</li>
`;
